### PR TITLE
Implement rpc-post-v1-05

### PR DESCRIPTION
**Pull request by AkhilaIlla**

### CODE REVIEW

File: common/changes/@microsoft.azure/openapi-validator-rulesets/implement_RPC-POST-V1-05_2023-02-15-21-56.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft.azure/openapi-validator-rulesets",
+      "comment": "",
```
Comment (jianyexi) [inline common/changes/@microsoft.azure/openapi-validator-rulesets/implement_RPC-POST-V1-05_2023-02-15-21-56.json:5]: comment should not be none.

File: packages/rulesets/src/spectral/functions/parameters-in-post.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    return []
+  }
+
+  if (postParameters.length === 0) {
+    return []
+  }
+
+  const path = ctx.path
+  const queryParams = postParameters.filter((param) => param.in === "query" && param.name !== "api-version")
+  const errors = queryParams.map((param) => {
```
Comment (rkmanda) [inline packages/rulesets/src/spectral/functions/parameters-in-post.ts:12]: >queryParams [](http://example.com/codeflow?start=17&length=11)

should u not be checking the length here?

File: packages/rulesets/src/spectral/test/parameters-in-post.test.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+          responses: {
+            200: {
+              description: "Success",
+            },
+          },
+        },
+      },
+    },
+    parameters: {
+      LoadTestNameParameter: {
```
Comment (rkmanda) [inline packages/rulesets/src/spectral/test/parameters-in-post.test.ts:35]: >LoadTestNameParameter [](http://example.com/codeflow?start=6&length=21)

unused?

Comment (rkmanda) [reply packages/rulesets/src/spectral/test/parameters-in-post.test.ts:35]: was it intentional?

File: packages/rulesets/src/spectral/test/parameters-in-post.test.ts [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+import linterForRule from "./utils"
+
+let linter: Spectral
+
+beforeAll(async () => {
+  linter = await linterForRule("ParametersInPost")
+  return linter
+})
+
+test("ParametersInPost should find errors", () => {
```
Comment (rkmanda) [inline packages/rulesets/src/spectral/test/parameters-in-post.test.ts:11]: >ParametersInPost [](http://example.com/codeflow?start=6&length=16)

some more tests to consider

1. More than  one query params which are not api-version
2. Query params with only api-version should not flag errors

**Review by rkmanda — CHANGES_REQUESTED** [review]
:clock1:

File: packages/rulesets/src/spectral/test/parameters-in-post.test.ts [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+      "loadTestName is a query parameter. Post operation must not contain any query parameter other than api-version."
+    )
+    expect(results[1].path.join(".")).toBe("paths./foo.post.parameters")
+    expect(results[1].message).toContain(
+      "quotaBucketName is a query parameter. Post operation must not contain any query parameter other than api-version."
+    )
+  })
+})
+
+test("ParametersInPost should not flag error for api-version param", () => {
```
Comment (rkmanda) [inline packages/rulesets/src/spectral/test/parameters-in-post.test.ts:117]: >should not flag error for api-version param [](http://example.com/codeflow?start=23&length=43)

should flag error for other query params but should not flag error for api-version param

**Review by rkmanda — APPROVED** [review]
:shipit:

**Comment by jianyexi** [comment]
@AkhilaIlla could you fix the CI test failure?

**Pull request merged**